### ISSUE TITLE
Resizing distance between levels? 

### ISSUE THREAD

**Issue by JoshuaCrotts** [abego/treelayout#5]
Hey, I'm working on a semantic tableaux generator, and one characteristic of it is having nodes that stack, and those that branch (i.e., if a node has only one child, its children are placed directly under, and if a node has two, it's like a standard binary tree). Also, when nodes stack, they don't have any lines between them. Now, I've got this set up in my program, but it looks a little weird since the stacked nodes are too far apart, but the branched nodes are too close. If you'd like a picture of what I'm talking about, I'd be glad to share.

For the record, what an amazing library. No wonder ANTLR uses it!

**Comment by abego** [comment]
Hi Joshua,

From your description I cannot tell what the problem is.
Maybe you can send me a small test program that demonstrate the issue and I will have a look. Also a sketch how you think it should look like would be fine.

Udo

**Comment by JoshuaCrotts** [comment]
Hi Udo,

So, here's what I'm after:

<img width="404" alt="Screen Shot 2021-04-10 at 5 33 16 PM" src="https://user-images.githubusercontent.com/18061804/114285152-dcdd7a80-9a22-11eb-9d1c-063b3989cdd1.png">

And here's what I currently have:

<img width="395" alt="Screen Shot 2021-04-10 at 5 31 32 PM" src="https://user-images.githubusercontent.com/18061804/114285135-b3245380-9a22-11eb-929d-384a370bcc83.png">

Without getting too into the specifics on truth trees, notice how some trees have two children, and others only have one. The former are called branching nodes, and the latter are stacking nodes. What I want is to be able to _increase_ the vertical distance between a node that branches and its two children, while _decreasing_ the space between a node that stacks and its child. Right now, the distances are uniform (because of the configuration), but I was wondering if there's a way to change it internally in Abego? Obviously, I could use offsets in JavaFX but that would conflict with how the tree is set up under the hood.

**Comment by JoshuaCrotts** [comment]
Here's a better illustration of which specific vertical spaces I'm trying to adjust:

![sketchpad pro_D68BBFE0E999174E393(1)](https://user-images.githubusercontent.com/18061804/114285256-d7346480-9a23-11eb-963b-01b794d9c366.png)

**Comment by abego** [comment]
Could there be branching nodes with a stacking node as parent, or will stacking nodes always be leaves?

**Comment by JoshuaCrotts** [comment]
In my example, the root is a branching node, and its right child is also a branching node (A -> (C & D)). The rest are stacking nodes. So, to answer your question, yes, a node that stacks could have a child that branches. 

Here's a good example. The root stacks its two children and both branch. Note that the underlying system I have knows when a tree branches or stacks - I just need to be able to increase or decrease the vertical distance based on that (I don't really want to confuse/bore you with the semantics).

<img width="508" alt="Screen Shot 2021-04-10 at 6 48 38 PM" src="https://user-images.githubusercontent.com/18061804/114286512-5ed2a100-9a2d-11eb-8ce3-c3a65d330917.png">

**Comment by abego** [comment]
I had a look at the paper the TreeLayout implementation is based on. As it turns out the algorithm to layout the tree as compact as possible requires that all nodes of a given level are all placed also on one level visually. This 
restricts us when using different vertical distances between nodes. From what I learned from your examples with these restrictions we cannot implement the semantic tableaux as you had it in mind.

However... I came up with a solution that may be sufficient and allows you to use the abego Treelayout framework.

Have a look at this picture, created with the SwingDemo included in the TreeLayout project:
![semver](https://user-images.githubusercontent.com/696535/114318111-cced9580-9b0b-11eb-8325-5bb07740a0f9.png)

I think this comes quite close to what you are looking for. The basic idea is to merge all stacking nodes into one node, just as "multi line text". 

When I show the boxes around the nodes you will easily see what I mean:
![semver-box](https://user-images.githubusercontent.com/696535/114318140-ec84be00-9b0b-11eb-8450-25fcf972767f.png)

Compare this to the "standard" layout/tree:
![semver2](https://user-images.githubusercontent.com/696535/114318184-2d7cd280-9b0c-11eb-91e9-87d999d44665.png)

The sample code is committed. Have a look at the "SwingDemo" and run it with the arguments

    `semtab --nobox`
    
I hope this helps,

Udo
